File: src/dataset/python/clock/gemini/GeminiFlash_clock_v1.py
class Clock:
    __slots__ = ('h', 'm')

    def __init__(self, hour, minute):
        total_minutes = (hour * 60 + minute) % (24 * 60)
        self.h = total_minutes // 60
        self.m = total_minutes % 60

    def __repr__(self):
        return "{:02d}:{:02d}".format(self.h, self.m)

    def __eq__(self, other):
        return self.h == other.h and self.m == other.m

    def __add__(self, minutes):
        total_minutes = (self.h * 60 + self.m + minutes) % (24 * 60)
        hour = total_minutes // 60
        minute = total_minutes % 60
        return Clock(hour, minute)

    def __sub__(self, minutes):
        total_minutes = (self.h * 60 + self.m - minutes) % (24 * 60)
        hour = total_minutes // 60
        minute = total_minutes % 60
        return Clock(hour, minute)

File: src/dataset/go/clock_Exercism_go_drapala/gemini/GeminiFlash_clock_v1.go
package clock

import "fmt"

// Clock represents a time of day in hours and minutes.
type Clock struct {
	hour, minute int
}

// String returns the time in HH:MM format.
func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hour, c.minute)
}

// New creates a new Clock.
func New(hour int, minute int) Clock {
	totalMinutes := hour*60 + minute
	return normalize(totalMinutes)
}

// Add adds minutes to the clock.
func (c Clock) Add(minutes int) Clock {
	return normalize(c.hour*60 + c.minute + minutes)
}

// Subtract subtracts minutes from the clock.
func (c Clock) Subtract(minutes int) Clock {
	return normalize(c.hour*60 + c.minute - minutes)
}

// normalize converts total minutes to a Clock.
func normalize(totalMinutes int) Clock {
	totalMinutes %= (24 * 60)

	if totalMinutes < 0 {
		totalMinutes += (24 * 60)
	}

	hour := totalMinutes / 60
	minute := totalMinutes % 60

	return Clock{hour, minute}
}